jade/jobs/job_configuration.py

```diff
@@ -50,7 +50,7 @@ def __init__(
         container : JobContainerInterface
 
         """
-        self._jobs = container or JobContainerByKey()
+        self._jobs = container or JobContainerByName()
```

Comment:
We used to store jobs by a key where each key had encodings that were meaningful for searching and filtering. That is no longer the case. We rely on unique job names. Somewhere along the way we forgot to change this. It is important because `get_job` was searching a list, comparing by name, instead of directly looking up the name in a dict.